bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
I think you are making things unnecessarily difficult -- setting at initialization in this case really just means "can be initialized in a single line". I understand why you don't want to pass *changing* global  state around, but the "never changed once initialized" semantics seems a reasonable distinction to me.

Still, your code base, so I will change it that way, once I have fixed the lisp problem.